DiscoveryInsightsinEinsteinAnalytics.apxc

```diff
@@ -0,0 +1,140 @@
+@RestResource (urlMapping = '/DiscoveryInsights' )
+global class EAPlus
+{
+    @HttpPost
+    global static string DiscoveryInsights(String relatingTo, String storyID, string relatingToDimension, string insights, string changingTo)
+    {
+        List<Response> result = new  List<Response>();
+                
+        Http http = new  Http();
+        HttpRequest httpReq = new  HttpRequest();
+        HttpResponse httpRes = new  HttpResponse();
+        //change this to your org endpoint
+        httpReq.setEndpoint('https://domain.my.salesforce.com/services/data/v45.0/smartdatadiscovery/proxy');
```

Comment:
You should be able to use `https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_url.htm#apex_System_URL_getSalesforceBaseUrl` to get the base path without hardcoding it.